platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/SignatureValidator.java

```diff
@@ -99,7 +114,24 @@ private boolean isValidSignature(final BaseEvent event, final PublicKey publicKe
     @Override
     public boolean isEventValid(final GossipEvent event) {
         final NodeId creatorId = event.getHashedData().getCreatorId();
-        final PublicKey publicKey = keyMap.get(creatorId);
+        final SoftwareVersion eventSoftwareVersion = event.getHashedData().getSoftwareVersion();
+        final int softwareComparison = currentSoftwareVersion.compareTo(eventSoftwareVersion);
+        final PublicKey publicKey;
+        if (softwareComparison < 0) {
+            // current software version is less than event software version
+            logger.error(
+                    EXCEPTION.getMarker(),
+                    "Cannot validate events for software version {} that is greater than the current software version {}",
+                    eventSoftwareVersion,
+                    currentSoftwareVersion);
```

Comment:
I suggest we throttle this log. An attacker could intentionally send events that might cause us to invoke this thousands of times per second.

Reply:
rate limited the log.